Remotion/Data/DomainObjects/DataManagement/RelationEndPoints/ExistingDataContainerEndPointsRegistrationAgent.cs

```diff
@@ -60,17 +59,20 @@ protected override string GetUnregisterProblem (IRelationEndPoint endPoint, Rela
       // (This only affects 1:n relations: for those, the opposite virtual end-point can be changed although the (one of many) real end-point is 
       // unchanged. For 1:1 relations, the real and virtual end-points always have an equal HasChanged flag.)
 
-      var maybeOppositeEndPoint =
-          Maybe
-              .ForValue (endPoint as IRealObjectEndPoint)
-              .Select (ep => RelationEndPointID.CreateOpposite (ep.Definition, ep.OppositeObjectID))
-              .Select (oppositeEndPointID => relationEndPointMap[oppositeEndPointID]);
-      if (maybeOppositeEndPoint.Where (ep => ep.HasChanged).HasValue)
+      var realObjectEndPoint = endPoint as IRealObjectEndPoint;
+
+      if (realObjectEndPoint == null)
+        return null;
+
+      var oppositeEndPointID = RelationEndPointID.CreateOpposite (realObjectEndPoint.Definition, realObjectEndPoint.OppositeObjectID);
+      var oppositeEndPoint = relationEndPointMap[oppositeEndPointID];
+
+      if (oppositeEndPoint?.HasChanged ?? false)

```

Comment:
Interesting. Yes, I like the new way of doing things better, wouldn't have realized that oppositeEndPoint can be null from reading the original code (while not remembering all details on MayBe).